postgresql/resource_postgresql_role.go

```diff
@@ -644,9 +694,48 @@ func setRoleValidUntil(db *sql.DB, d *schema.ResourceData) error {
 
 	roleName := d.Get(roleNameAttr).(string)
 	sql := fmt.Sprintf("ALTER ROLE %s VALID UNTIL '%s'", pq.QuoteIdentifier(roleName), pqQuoteLiteral(validUntil))
-	if _, err := db.Exec(sql); err != nil {
+	if _, err := txn.Exec(sql); err != nil {
 		return errwrap.Wrapf("Error updating role VALID UNTIL: {{err}}", err)
 	}
 
 	return nil
 }
+
+func revokeRoles(txn *sql.Tx, d *schema.ResourceData) error {
+	role := d.Get(roleNameAttr).(string)
+
+	query := `SELECT array_remove(array_agg(role_name::text), NULL)
```

Comment:
I've probably copied the query and in this case the ``array_agg`` is not necessary as, as you said,  I need to loop anyway.

For the big query in the read method, ``array_agg`` is needed and array_remove aswell because, with a left join, if there is no matching rows in the right table ``array_agg`` will produce ```{NULL}```